### PR TITLE
Add support for macOS fcntl(F_FULLFSYNC) (closes #6)

### LINKED ISSUE #6: Add support for macOS fcntl(F_FULLFSYNC)
This has the effect of saving a ~quarter of Percona Server 5.7 testsuite run time, Apple consumer SSD, by the virtue of no-opping MySQL my_sync function. I should be able to submit a patch.

### PR DESCRIPTION
**Pull request by laurynas-biveinis**
On macOS, replace fcntl syscall, check if its called for F_FULLFSYNC,
and make it a no-op. Add a new test fcntltest.c, verify with dtruss
that F_FULLFSYNC does not happen.

At the same time add missing entries to .gitignore, and mark
sync_file_range as LIBEATMYDATA_API

### CODE REVIEW

File: libeatmydata/libeatmydata.c [hunk 1]
```diff
@@ -236,3 +246,30 @@ int sync_file_range(int fd, off64_t offset, off64_t nbytes, unsigned int flags)
...
+#if defined(F_FULLFSYNC) && defined(__APPLE__)
+
+/* fcntl does not have a va_list version. We have to trust this one seen in
+public Darwin sources:
+int __FCNTL(int, int, void *); */
+
+int LIBEATMYDATA_API fcntl(int fd, int cmd, ...)
+{
+	if ((eatmydata_is_hungry() && (cmd == F_FULLFSYNC))) {
+		pthread_testcancel();
```
Comment (laurynas-biveinis) [inline libeatmydata/libeatmydata.c]: This one might be redundant given that we call actual fcntl on the next line.

Comment (stewartsmith) [reply libeatmydata/libeatmydata.c]: Probably is, I'd expect that if it's a cancellation point that fcntl would take care of it for other fctnl commands.

Comment (stewartsmith) [reply libeatmydata/libeatmydata.c]: anyway, i can fix that in merging.

**Comment by stewartsmith** [comment]
With the fcntl change, merged as of 1e5ae05

**Pull request closed without merging**